### ISSUE TITLE
Unintuitive behavior when both features and package are specified

### ISSUE THREAD

**Issue by kornelski** [rust-lang/cargo#8368]
If I have a workspace:

```text
Cargo.toml
foo/Cargo.toml
```

and run in the root of the workspace:

```sh
cargo build -p foo --features=bar
```

Cargo fails with an error #7742:

> error: --features is not allowed in the root of a virtual workspace

while

```sh
cd foo
cargo build --features=bar
```

is accepted. It's surprising that specifying `-p` is not equivalent to building in that package's directory. 

I expected that when `-p` points to a specific package, Cargo wouldn't try to use non-existent features from the root manifest (#4942), but instead look for the features in that package's own manifest.

**Comment by ehuss** [comment]
There is a solution for this implemented via `-Zpackage-features` (#5364).  `cargo build -p foo --features bar -Zpackage-features` should work.

**Comment by kornelski** [comment]
Perfect! Thank you.
